src/constructors/keyframes.js

```diff
@@ -5,24 +5,41 @@ import StyleSheet from '../models/StyleSheet'
 
 const replaceWhitespace = (str: string): string => str.replace(/\s|\\n/g, '')
 
+export class Keyframes {
+  id: string
+  name: string
+  rules: Array<string>
+
+  constructor(name: string, rules: Array<string>) {
+    this.name = name
+    this.rules = rules
+
+    this.id = `sc-keyframes-${name}`
+  }
+
+  inject(styleSheet: StyleSheet) {
+    if (!styleSheet.hasNameForId(this.id, this.name)) {
+      styleSheet.inject(this.id, this.rules, this.name)
+    }
+  }
+
+  toString() {
+    return this.name
```

Comment:
This is important for https://github.com/styled-components/styled-components/blob/master/src/models/ComponentStyle.js#L91 to insert animation name

Reply:
See below; let’s keep it explicit without coercion